packages/world/ts/encodeSystemCall.ts

```diff
@@ -1,13 +1,22 @@
 import { Abi, EncodeFunctionDataParameters, Hex, encodeFunctionData, type ContractFunctionName } from "viem";
 import type { AbiParametersToPrimitiveTypes, ExtractAbiFunction } from "abitype";
 import { worldCallAbi } from "./worldCallAbi";
+import { internal_normalizeSystemFunctionName } from "./normalizeSystemFunctionName";
 
-export type SystemCall<abi extends Abi, functionName extends ContractFunctionName<abi>> = EncodeFunctionDataParameters<
-  abi,
-  functionName
-> & {
+export type SystemCall<abi extends Abi, functionName extends ContractFunctionName<abi>> = {
+  /**
+   * System ABI
+   */
+  readonly abi: abi;
+  /**
+   * System's resource ID
+   */
   readonly systemId: Hex;
-};
+  /**
+   * System function name to call
+   */
+  readonly functionName: functionName;
```

Comment:
pulled these out for stronger types in the function body and so we can add TSDoc to clarify the `abi` param